common/utils/async_query_helpers.ts

```diff
@@ -41,6 +42,9 @@ export const executeAsyncQuery = (
           ...query,
           sessionId: getAsyncSessionId(currentDataSource) ?? undefined,
         }),
+        query: {
+          dataSourceMDSId: dataSourceMDSId,
```

Comment:
as suggested, we can use shorthand here
```suggestion
          dataSourceMDSId,
```